src/modules/0.0.18/aws_cloudfront/entity/distribution.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  Column,
+  Entity,
+  PrimaryGeneratedColumn,
+} from 'typeorm'
+
+import { cloudId, } from '../../../../services/cloud-id' // This is ridiculous. Can we fix this?
+
+export enum viewerProtocolPolicyEnum {
+  ALLOW_ALL = "allow-all",
+  REDIRECT_TO_HTTPS = "redirect-to-https",
+  HTTPS_ONLY = "https-only"
+}
+
+export enum originProtocolPolicyEnum {
+  HTTP_ONLY = "http-only",
+  MATCH_VIEWER = "match-viewer",
+  HTTPS_ONLY = "https-only"
+}
+
+@Entity()
+export class Distribution {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @Column({
+    nullable: true,
+  })
+  @cloudId
+  distributionId?: string;
+
+  @Column({
+    nullable: true,
+  })
+  callerReference?: string;
+
+  @Column({
+    nullable: true,
+  })
+  comment?: string;
+
+  @Column({
+    nullable: true,
+  })
+  enabled?: boolean;
+
+  @Column({
+    nullable: true,
+  })
+  isIPV6Enabled?: boolean;
+
+  @Column({
+    nullable: true,
+  })
+  webACLId?: string;
+
+  @Column({
+    type: 'json',
+    nullable: false,
+  })
+  defaultCacheBehavior: { TargetOriginId: string | undefined, ViewerProtocolPolicy: viewerProtocolPolicyEnum, CachePolicyId: string | undefined };
+
+  @Column({
+    type: 'json',
+    nullable: false,
+  })
+  origins: { DomainName: string | undefined, Id: string | undefined, OriginShield: any,
+    CustomOriginConfig: { HTTPPort: number|undefined, HTTPSPort: number|undefined, OriginProtocolPolicy: originProtocolPolicyEnum }
+  }[];
```

Comment:
1. A general note that the `type: 'json',` records have no constraints inside of the database itself, so even with the strong typing here in TS-land, there's nothing to prevent the user from setting things like `TargetOriginId: true`, so we should *definitely* think about how to eliminate these in the future.
2. It would be more legible to format these like this:

```suggestion
  @Column({
    type: 'json',
    nullable: false,
  })
  defaultCacheBehavior: {
    TargetOriginId: string | undefined,
    ViewerProtocolPolicy: viewerProtocolPolicyEnum,
    CachePolicyId: string | undefined
  };

  @Column({
    type: 'json',
    nullable: false,
  })
  origins: {
    DomainName: string | undefined,
    Id: string | undefined,
    OriginShield: any,
    CustomOriginConfig: {
      HTTPPort: number | undefined,
      HTTPSPort: number | undefined,
      OriginProtocolPolicy: originProtocolPolicyEnum
    }
  }[];
```